pylint/extensions/typing.py

```diff
@@ -124,6 +124,11 @@ class TypingChecker(BaseChecker):
             "Python 3.9.0 and 3.9.1. Use ``typing.Callable`` for these cases instead. "
             "https://bugs.python.org/issue42965",
         ),
+        "R6006": (
+            "Type-hint `%s` is used more than once in assignment. Consider removing redundant typehints. ",
+            "redundant-union-assign-typehint",
```

Comment:
```suggestion
            "duplicated-type-in-union",
```
(This is a suggestion, I'm trying to make the name clearer)

Reply:
maybe `duplicated-union_typehint` ?

Reply:
``duplicated-typehint-in-union`` ?

Reply:
How about we make it generic and similar to what the docs say:
`redundant-typehint-argument`

[_Redundant arguments are skipped_](https://docs.python.org/3/library/typing.html#typing.Union)

Reply:
@mbyrnepr2  - nice idea to check it out mate.  Do you suggest it both for assigns and functional arguments? 
I will explain myself - This checker only covers assignments for now. After this is closed I would extend it for function arguments, do you think they both should have the same message?

Reply:
Yeah I think the name of the message should be the same in all cases. So, if the duplication occurs in an assignment or function parameter etc. it is the same issue in all cases.